lib/components/tabs/tabs.dart

```diff
@@ -0,0 +1,136 @@
+import 'dart:html';
+import 'dart:async';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('Tabs');
+
+class TabView {
+  String id;
+  String label;
+  DivElement content;
+
+  TabView(this.id, this.label, this.content);
+}
+
+class TabsView {
+  DivElement renderElement;
+  List<TabView> _tabs;
+  Map<String, TabView> _tabsLookup;
+
+  String _selectedTabID;
+  String _tabContentClassname;
+
+  List<SpanElement> _tabChoosers = [];
+  DivElement _tabsHeader;
+  DivElement _tabContent;
+
+  Stream<String> _onEmpty;
+  StreamController<String> _onEmptyController;
+  Stream<String> get onEmpty => _onEmpty;
+
+  TabsView(this._tabs, {String defaultSelectedID, String tabContentClassname}) {
+    _updateLookup();
+
+    if (defaultSelectedID != null) {
+      _selectedTabID = defaultSelectedID;
+    } else {
+      _selectedTabID = _tabs.first.id;
+    }
+
+    _tabContentClassname = tabContentClassname;
+
+    _tabsHeader = DivElement();
+    _tabs.forEach((tab) {
+      var tabChooser = _getTabChooser(tab);
+      _tabChoosers.add(tabChooser);
+      _tabsHeader.append(tabChooser);
+    });
+
+    _tabContent = DivElement()
+      ..className = "tab-content"
+      ..append(_tabsLookup[_selectedTabID].content);
+    if (_tabContentClassname != null) {
+      _tabContent.classes.add(_tabContentClassname);
+    }
+
+    renderElement = DivElement()..append(_tabsHeader)..append(_tabContent);
+
+    this._onEmptyController = StreamController();
+    this._onEmpty = _onEmptyController.stream;
+  }
+
+  SpanElement _getTabChooser(TabView tab) {
+    var tabChooser = SpanElement()
+      ..innerText = tab.label
+      ..className = "tab-chooser"
+      ..id = "tab-chooser-${tab.id}"
```

Comment:
I'd prefer us to use the `dataset` property on the element for storing the ID, rather than the CSS ID, which tends to be used more for formatting and styling than for storing data:
```suggestion
      ..dataset['id'] = tab.id
```